src/components/navigation/z-user-dropdown/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { Component, Prop, State, h } from "@stencil/core";
+import { MenuItem, ThemeVariant, ThemeVariantBean } from "../../../beans";
+import { mobileBreakpoint } from "../../../constants/breakpoints";
+
+@Component({
+  tag: "z-user-dropdown",
+  styleUrl: "styles.css",
+  shadow: true,
+})
+export class ZUserDropdown {
+  /** logged status flag */
+  @Prop() logged?: boolean;
+  /** user full name */
+  @Prop() userfullname?: string;
+  /** Json stringified or array to fill menu dropdown */
+  @Prop() menucontent?: string | MenuItem[];
+  /** theme variant, default 'dark' */
+  @Prop() theme?: ThemeVariantBean = ThemeVariant.dark;
+
+  @State() ismenuopen: boolean = false;
+
+  private linkarray: MenuItem[];
+
+  componentWillRender() {
+    this.linkarray =
+      typeof this.menucontent === "string"
+        ? JSON.parse(this.menucontent)
+        : this.menucontent;
+  }
+
+  handleToggle() {
+    this.ismenuopen = !this.ismenuopen;
+  }
+
+  renderCaretIcon() {
+    const direction = this.ismenuopen ? "up" : "down";
+    return (
+      <z-icon
+        slot="login"
```

Comment:
Is slot necessary here? You already fit the whole z-user-dropdown inside z-app-topbar

Reply:
Moreover, do we consider correct to specify a named slot like this? Surely z-app-topbar and z-user-dropdown are tightly coupled, but shouldn't z-user-dropdown be reusable outside of z-app-topbar (even if we know it will never be the case)?

Reply:
I answer both: inserting a component in a slot rather than inside another component helps to handle events and accessibility because the slotted component shares the same DOM tree with the host component (and with the whole app that hosts both). 
The z-app-topbar component can also work without z-user-dropdown as a login component, but the presence of a "login  block/feature" in the topbar is foreseen by design (https://app.abstract.com/projects/fd370780-e659 -11e8-99dc-0d08537c5fde / branches / a870e9e3-ef17-444f-A2B2-e25e48784735 / commits / b683534800ef558c7e336bf60a1d397eaa4c66cb / files / 7e5534c1-96e2-44c2-98f8-965c34ee8835 / layers / 7BC1646B-F8EC-4466-A720-AEBAFFC3ABF0? CollectionID = f3a099d7 -cda4-4eaf-9bcd-5ecaa786a7e9 & collectionLayerId = 8578e239-e088-44b4-bbc8-203eabde74e2 & mode = design). 
the z-user-dropdown component can also be used outside the topbar, although by now there are no other use cases that specify its intrinsic characteristics

Reply:
The point I want to make is, I am not aspecting to find a named slot specified like this. Even if you can surely reuse this component, its definition is coupled with z-app-topbar.
I am expecting to use a named slot only in the app exploing the Albe Library

Reply:
ok @Ocram83 and @saldomik my fault, I misunderstand the comment. fixed in next commit